plugins/package-managers/node/src/main/kotlin/PackageJson.kt

```diff
@@ -49,9 +50,32 @@ internal fun parsePackageJson(file: File): PackageJson = parsePackageJson(file.r
 
 internal fun parsePackageJson(json: String): PackageJson = parsePackageJson(JSON.parseToJsonElement(json))
 
-internal fun parsePackageJsons(jsons: String): List<PackageJson> =
+internal fun parsePackageJsons(jsons: String): List<Result<PackageJson>> =
     jsons.byteInputStream().use { input ->
-        JSON.decodeToSequence<JsonElement>(input).mapTo(mutableListOf()) { parsePackageJson(it) }
+        JSON.decodeToSequence<JsonElement>(input).mapTo(mutableListOf()) { element ->
+            runCatching {
+                parsePackageJson(element)
+            }.recoverCatching {
+                if (element !is JsonObject) throw it
+
+                // Try to enrich the exception with more package details that are still available.
```

Comment:
In my view this adds to much code for just an error message. 
I believe normally it would suffice to have the error say:
"Could not parse package.json for modulle $moduleId". In particular because
this is a rare edge case which should not happen.
Such error message would provide sufficient information for debugging.

...so, the exception could be could in the caller and such simple error could be logged. However, as we have this  `parsePackageJsons()` operating on multiple package.json, this is not easily possible. 
While I prefer to refactor the code to make this possible, as alternative I would propose to simply log the entire `package.json` string, e.g. `element.toString()`. 

What do you think?